### PR TITLE
Change type_suffix from @xxx to -xxx.

### PR DESCRIPTION
**Pull request by winksaville**
Using an '@' in a path in linker scripts is not allowed, and I can
imagine it causing problems with other programs. So I suggest
using dash, '-', rather than '@' in the type_suffix() methods.

As an example in baremetal-hi/link.meson.ld, the current solution that
Jussi provided was to use an asterisk, '*', in the file name see:
winksaville/baremetal-hi@d358e06

But using an asterisk to resolve the problem could cause multiple matches
leading to unforeseen errors.

### CODE REVIEW

**Comment by jpakkane** [comment]
I had some sort of a git mixup with that PR. It should have used '?' instead of '*'. The question mark only matches one character so it is the correct solution there.

The private directory naming scheme is more complicated than first appears. As an example we want the private directories to be immediately recognizable in a directory listing so they are immediately recognizable. There are several different separator characters used (yes, one of them is the dash) and these need to be different so two different things don't get aliased to the same thing.

I'm closing this MR for now. That does not mean that we can't change the naming scheme, but any change must be prepared with the entire naming scheme in mind.

Thanks,

**Comment by winksaville** [comment]
OK, I do think it should be changed and I'd be glad to look at it if you want, but I'll need to be pointed in the right direction. Maybe there's a wiki page or ...

**Comment by jpakkane** [comment]
Determining an internal name mangling like this one is ultimately a matter of taste. There are a million ways to do it and people prefer different solutions and you can waste an unlimited number of hours debating how to do it.

We're not going to do that. Instead this needs to be an actual use case. That is, someone needs to produce a real world example where the current mangling causes an actual error (unable to compile, breaks some tools etc). At that point we can design a new one to fix the issue. Note that the current issue is not a sufficient reason because a ? is a straightforward solution and ld's script format is not very elegant and you have to use wildcards on it quite often anyway.

tl/dr: unless there is an actual demonstrable bug, let's focus on more important things instead.

Thanks,

**Comment by winksaville** [comment]
OK, you're the master :)
